### PR TITLE
Update Woo Blaze Jetpack connection copy

### PR DESCRIPTION
**Pull request by Alameen688**
## Proposed Changes

* This PR updates the connect auth and sign-up page copy for Woo Blaze

## Testing Instructions

* Create a WooCommerce site, and make sure the Jetpack plugin is deactivated or not installed
* Point `widgets.wp.com` to your WP Sandbox.
* Run `yarn start` to start calypso 
* Create a plugin build from the woo-blaze plugin repo and install it in your test blog OR run locally as described in repo readme.
* cd `apps/blaze-dashboard` and `yarn dev --sync` to sync blaze dashboard to WP sandbox OR run the commands here https://github.com/Automattic/wp-calypso/pull/90070#issuecomment-2088451908
* In WP Admin, Go to `Marketing` -> `Blaze for WooCommerce`
* Click `Connect now` button and you should see the Jetpack connect flow.

For existing connected users you should see the following connect screen:
sample-connect

For a new user/user without a previous connection attempt you should see the following connect screens:
new user flow
email confirmationemail content

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by sbarbosa** [comment]
Hi @Alameen688! 
We have a PR almost ready with the changes of the new connection page https://github.com/Automattic/wp-calypso/pull/90160. 

Checking this PR; I think we will need to revert some changes to adapt them to that other PR. I think that the only changes we need to maintain are:
* Changes to the jetpack-client code
* Redirection using the `connect_url` config prop

From our last sync, I think the changes to the connection-info URL are not necessary, because the redirection will be to the backend, and not the frontend.

We can wait until we merged that other PR, and then revert back the changes that are outdated. Let me know what you think.

**Comment by Alameen688** [comment]
> I think we will need to revert some changes to adapt them to that other PR

Yes, I agree. Those changes were just a placeholder for testing pending the implementation of the connect UI. And yes, we don't need the connection-info URL it was under the assumption that the new connect UI on the dashboard might be a separate page but I can see in the other PR https://github.com/Automattic/wp-calypso/pull/90160 that it is a component conditionally rendered under the setup page/URL. 

_Also, a long explanation for the connection-info page - as I mentioned in our sync is that it doesn't do anything in this PR it's just an empty placeholder page for when the page reloads to trigger the serverside logic that redirects to the jetpack connection page. the connect URL can be the setup URL or home URL (i.e. just the `wc-blaze` path), just needed a URL to trigger the onboarding logic which is handled server side, it could have been a `window.reload` on the client but we just have it as a `connect_url` to give control/flexibility to the server of what happens when the connect button is clicked in case of a change in the redirect flow, so client-side code changes won't be needed_

File: apps/blaze-dashboard/src/pages/setup/components/blaze-disabled.jsx
Comment (Alameen688) [inline apps/blaze-dashboard/src/pages/setup/components/blaze-disabled.jsx]: NB: The Connect button and related component and style will be removed in favor of https://github.com/Automattic/wp-calypso/pull/90160 but leaving it here temporarily for easy testing until that PR gets merged and the changes made available to this PR.

**Review by sbarbosa — APPROVED** [review]
Thanks for the changes @Alameen688. 
The code looks great!

File: client/jetpack-connect/signup.js
Comment (Alameen688) [inline client/jetpack-connect/signup.js:409]: Using this arg and not hardcoding the name "Blaze for WooCommerce". just so the main copy section can be reused across services if needed. _It's same as the one above it used by WooPayments and can be shared later if necessary (after translation)_

**Review by therocket-gr — APPROVED** [review]
tested and works as described
code changes LGTM

**Pull request merged**
